app/models/framework_response.rb

```diff
@@ -32,7 +32,10 @@ def update_with_flags!(value)
 
       update!(flags: build_flags)
       clear_dependent_values_and_flags!(old_value)
+      person_escort_record.update_state!
     end
+  rescue FiniteMachine::InvalidStateError
+    raise ActiveRecord::ReadOnlyRecord, "Can't update framework_responses because person_escort_record is #{person_escort_record.state}"
```

Comment:
If an attempt to update a response when the PER is confirmed/printed, a forbidden error is raised for the record being read only now